components/automate-ui/src/app/pages/project/details/project-details.component.ts

```diff
@@ -118,28 +120,26 @@ export class ProjectDetailsComponent implements OnDestroy {
     this.deleteModalVisible = false;
   }
 
-  startRuleDelete(r: any): void {
+  startRuleDelete(rule: Rule): void {
     this.deleteModalVisible = true;
-    this.ruleToDelete = r;
+    this.ruleToDelete = rule;
   }
 
   deleteRule(): void {
-    this.store.dispatch(new DeleteRule({id: this.ruleToDelete.id}));
+    this.store.dispatch(new DeleteRule({ id: this.ruleToDelete.id }));
     this.closeDeleteModal();
   }
 
-  getEditStatus(status: string): string {
-    return status === 'staging'
-      ? 'Edits pending'
-      : 'Applied';
+  getEditStatus(rule: Rule): string {
+    return rule.status === 'staged' ? 'Edits pending' : 'Applied';
   }
 
-  showDeleteRule(rule: Rule): boolean {
-    return rule.edits !== 'staging';
+  showDeleteRule(): boolean {
+    return true; // TODO: return false when *project* status is "updating..."
   }
 
   showProjectLink(): boolean {
-    return _find(this.rules, ['edits', 'staging']) ? true : false;
+    return some(['status', <RuleStatus>'staged'], this.rules);
```

Comment:
❓ The `<RuleStatus>` is not inferred? Hmm!

Reply:
I investigated this and other use cases of type casting. Will post some notes in Slack on Monday...